webhook.go

```diff
@@ -0,0 +1,178 @@
+package webhook
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"strings"
+
+	kwhhttp "github.com/slok/kubewebhook/pkg/http"
+	kwhlog "github.com/slok/kubewebhook/pkg/log"
+	kwhmutating "github.com/slok/kubewebhook/pkg/webhook/mutating"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// binVolumeName is the name of the volume where the SecretHub CLI binary is stored.
+	binVolumeName = "secrethub-bin"
+
+	// binVolumeMountPath is the mount path where the SecretHub CLI binary can be found.
+	binVolumeMountPath = "/secrethub/bin/"
+)
+
+// binVolume is the shared, in-memory volume where the SecretHub CLI binary lives.
+var binVolume = corev1.Volume{
+	Name: binVolumeName,
+	VolumeSource: corev1.VolumeSource{
+		EmptyDir: &corev1.EmptyDirVolumeSource{
+			Medium: corev1.StorageMediumMemory,
+		},
+	},
+}
+
+// binVolumeMount is the shared volume mount where the SecretHub CLI binary lives.
+var binVolumeMount = corev1.VolumeMount{
+	Name:      binVolumeName,
+	MountPath: binVolumeMountPath,
+	ReadOnly:  true,
+}
+
+// SecretHubMutator is a mutator.
+type SecretHubMutator struct {
+	logger kwhlog.Logger
+}
+
+// Mutate implements MutateFunc and provides the top-level entrypoint for object
+// mutation.
+func (m *SecretHubMutator) Mutate(ctx context.Context, obj metav1.Object) (bool, error) {
```

Comment:
Can we be sure that this is only ran once for every Pod?

Reply:
I believe that within a single request, `Mutate` is [called only  once](https://github.com/slok/kubewebhook/blob/v0.10.0/pkg/webhook/mutating/webhook.go#L128).

And because mutation starts from the pod specification that is sent with the request, it is idempotent between failed requests. Sending the same request twice doesn't lead to a different response.

Why do you ask?

Reply:
I aked because I've also seen mutating webhooks that store in an annotation whether they have executed. So I wanted to make sure this thing is idempotent. But if you're sure that it is always called with the original spec as input, it sounds like it is OK.

Reply:
Ah. I've seen a similar [example](https://github.com/slok/kubewebhook/tree/v0.10.0/examples/pod-annotate). I believe this annotation is an example and a way to show the mutator has done something and is not part of any logic that provides idempotency.